src/result-async.ts

```diff
@@ -0,0 +1,70 @@
+import { Result, Ok, Err } from './'
+
+export class ResultAsync<T, E> {
+  private _promise: Promise<Result<T, E>>
+
+  constructor(res: Promise<Result<T, E>>) {
+    this._promise = res
+  }
+
+  static fromPromise<T, E>(promise: Promise<T>, errorFn?: (e: unknown) => E): ResultAsync<T, E> {
+    let newPromise: Promise<Result<T, E>> = promise.then((value: T) => new Ok(value))
+    if (errorFn) {
+      newPromise = newPromise.catch(e => new Err<T, E>(errorFn(e)))
+    }
+    return new ResultAsync(newPromise)
+  }
```

Comment:
Good idea !

It's a bit tricky though because `process.env` is only for `node` runtimes. Isn't this lib also for the browser ?

We could either add a condition to check if it's a browser environment (would we want to log in the browser as well ?).

We could also make the second argument mandatory.

Reply:
@paduc good point. I was thinking that `process.env` would be available because of webpack or parcel.

I think this warning should be logged anywhere. The reason I didn't want to log in production is because the user might already be dealing with promise rejections on their own. So it would be annoying to keep logging the warning even though it's irrelevant to them.

I don't think making it mandatory makes sense because users may prefer to catch the exception and return an `Err` themselves.

Reply:
Ok to recap, the solution would be:
- In node, only log outside of prod and test environments
- In the browser, always log ?

What if we only log if the second argument is `undefined` (meaning, no warning if the user provides `null` for instance ?

Reply:
Yes, let's do that. Log outside of tests or prod for node. Always log in a browser environment (although some front end apps might have `process.env` defined, like webpack projects).

Don't add the `undefined` vs `null` functionality. I don't think it's worth it.